### PR TITLE
Add basic mappings

### PR DESCRIPTION
**Pull request by onbjerg**
Indexes pools, pairs, tokens, stakes and rewards. The subgraph does not currently index APY or number of poolers.

APY will be added in a separate PR and any other information that we do not need currently can also be added later.

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -14,7 +14,6 @@ type Token @entity {
   symbol: String!
   name: String!
   decimals: BigInt!
-  totalSupply: BigInt!
```
Comment (rperez89) [inline schema.graphql:17]: we are not going to use it for anything?

File: src/factory.ts [hunk 2]
```diff
@@ -1,6 +1,72 @@
...
+} from '../generated/Factory/Pool'
+import {
+  Pool as PoolDataSource
+} from '../generated/templates'
+import {
+  Pool
+} from '../generated/schema'
+import {
+  ZERO_BD,
+  loadOrCreateFactory,
```
Comment (rperez89) [inline src/factory.ts:19]: with fabri generally we try to alphabetize the imports :)

File: src/pool.ts [hunk 3]
```diff
@@ -1,22 +1,53 @@
...
 } from '../generated/templates/Pool/Pool'
+import {
+  Pool,
+  Token
+} from '../generated/schema'
+import {
+  exponentToBigDecimal
+} from './helpers'
+
+const EIGHTEEN_BI = BigInt.fromI32(18)
```
Comment (rperez89) [inline src/pool.ts:19]: maybe we can rename this to EIGHTEEN_BN if we are trying to make reference to a big number

**Review by rperez89 — COMMENTED** [review]
LGTM!

File: schema.graphql (hunk 1 above)
Comment (onbjerg) [reply schema.graphql:17]: Not for the moment. It is more complicated to implement than what we are gaining (currently nothing) as some tokens have a dynamic total supply (i.e. Honey)

File: src/factory.ts (hunk 2 above)
Comment (onbjerg) [reply src/factory.ts:19]: I like having the constants on top :(

File: src/pool.ts (hunk 3 above)
Comment (onbjerg) [reply src/pool.ts:19]: There are two types of big numbers in subgraphs; big decimals and big integers. There are some constants in the code base that exist for both (e.g. `ZERO_BD` and `ZERO_BI`) so I just reused that naming scheme for consistency

**Pull request merged**